src/include/catalog/index_schema.h

```diff
@@ -190,35 +207,44 @@ class IndexSchema {
     std::vector<std::unique_ptr<parser::AbstractExpression>> FromJson(const nlohmann::json &j);
 
    private:
-    static constexpr uint32_t MASK_VARLEN = 0x00FFFF00;
-    static constexpr uint32_t MASK_NULLABLE = 0x00000080;
-    static constexpr uint32_t MASK_TYPE = 0x0000007F;
-    static constexpr uint32_t OFFSET_VARLEN = 8;
+    bool ShouldHaveTypeModifier() const {
+      return (attr_length_ == storage::VARLEN_COLUMN &&
+              (type_ == type::TypeId::VARCHAR || type_ == type::TypeId::VARBINARY)) ||
+             (attr_length_ == 16 && type_ == type::TypeId::DECIMAL);
```

Comment:
If you do keep `attr_length` here, you may want to change `16` to `type::TypeUtil::GetTypeSize(type::TypeId::DECIMAL)` just so it's a bit clearer and also so it's easier to change the underlying type if we ever need to.